src/main/java/net/unicon/lti/service/harmony/HarmonyService.java

```diff
@@ -39,19 +41,31 @@ public class HarmonyService {
 
     RestTemplate restTemplate;
 
-    public HarmonyPageResponse fetchHarmonyCourses() {
+    public HarmonyPageResponse fetchHarmonyCourses(int page) {
 
         if (StringUtils.isAnyBlank(harmonyCoursesApiUrl, harmonyJWT)) {
             log.warn("The Harmony Courses API has not been configured, courses will not be fetched.");
             return null;
         }
 
         try {
+
             restTemplate = restTemplate == null ? new RestTemplate(new BufferingClientHttpRequestFactory(new SimpleClientHttpRequestFactory())) : restTemplate;
+
+            // Build the URL
+            String requestUrl = harmonyCoursesApiUrl;
+            // Only append the page parameter when it's needed.
+            if (page != 1) {
+                URIBuilder builder = new URIBuilder(harmonyCoursesApiUrl);
+                // Add the page parameter since this API is paginated.
+                builder.setParameter("page", String.valueOf(page));
+                requestUrl = builder.build().toString();
+            }
+
             HttpHeaders headers = new HttpHeaders();
             headers.setBearerAuth(harmonyJWT);
             HttpEntity<String> entity = new HttpEntity<>("parameters", headers);
-            ResponseEntity<HarmonyPageResponse> response = restTemplate.exchange(harmonyCoursesApiUrl, HttpMethod.GET, entity, HarmonyPageResponse.class);
+            ResponseEntity<HarmonyPageResponse> response = restTemplate.exchange(requestUrl, HttpMethod.GET, entity, HarmonyPageResponse.class);
```

Comment:
Could you please add a unit test to verify that if page equals 2, then the requestUrl value has the page parameter added to it as expected?

Reply:
Makes sense to me, I created a unit test to verify both the page number and the 'page' parameter name.